tasks/queue-granules/src/index.ts

```diff
@@ -0,0 +1,234 @@
+  const granules = (event.input.granules || []);
+  const memoizedFetchProvider = memoize(fetchGranuleProvider, (_, providerId) => providerId);
+  const memoizedFetchCollection = memoize(
+    collectionsApi.getCollection,
+    ({ collectionName, collectionVersion }) => constructCollectionId(
+      collectionName,
+      collectionVersion
+    )
+  );
+  const parentExecutionArn = buildExecutionArn(
+    get(event, 'cumulus_config.state_machine')!,
+    get(event, 'cumulus_config.execution_name')!
+  )!;
+  const pMapConcurrency = get(event, 'config.concurrency', 3);
+
+  const messageTemplate = await getJsonS3Object(
+    event.config.internalBucket,
+    templateKey(event.config.stackName)
+  );
+  const { arn: granuleIngestWorkflowArn } = await getJsonS3Object(
+    event.config.internalBucket,
+    getWorkflowFileKey(event.config.stackName, event.config.granuleIngestWorkflow)
+  );
+
+  const executionArns = await pMap(
+    createIterable(granules, event.config.preferredQueueBatchSize),
+    async ({ provider, collectionId, chunks }) => {
+      const { name: collectionName, version: collectionVersion } = deconstructCollectionId(
+        collectionId
+      );
+      const [collection, normalizedProvider] = await Promise.all([
+        memoizedFetchCollection({
+          prefix: event.config.stackName,
+          collectionName,
+          collectionVersion,
+        }),
+        memoizedFetchProvider(event, provider),
+      ]);
+
+      return await pMap(
+        chunks,
+        async (granuleBatchIn) => {
+          const granuleBatch = updateGranuleBatchCreatedAt(granuleBatchIn, Date.now());
+          await pMap(
+            granuleBatch,
+            (queuedGranule) => {
+              const { granuleId, updatedAt, createdAt } = queuedGranule;
+
+              if (updatedAt && (!Number.isInteger(updatedAt) || updatedAt < 0)) {
+                throw new Error(`Invalid collection information provided for granule with granuleId: ${granuleId}, please check task input to make sure collection information is provided`);
+              }
+              return granulesApi.updateGranule({
+                prefix: event.config.stackName,
+                collectionId,
+                granuleId,
+                body: {
+                  collectionId,
+                  granuleId,
+                  status: 'queued',
+                  // The non-null assertion operator is used below because we know
+                  // createdAt and updatedAt are non-null values given the condition
+                  // checking updatedAt above, and the function call to updateGranuleBatchCreatedAt
+                  // which will set createdAt.
+                  updatedAt: updatedAt!,
+                  createdAt: createdAt!,
+                },
+              });
+            },
+            {
+              concurrency: pMapConcurrency,
+              stopOnError: false,
```

Comment:
> Do their statuses get set to "failed", even in the cases where they were actually queued?
The statuses of the granules get set to `failed` which is correct. Though I think since this ticket is addressing converting queue granules to TS, I may leave this change for another ticket: https://bugs.earthdata.nasa.gov/browse/CUMULUS-3388 @chuckwondo

Reply:
Yep, makes perfect sense to address this separately.